client/resolve.c

```diff
@@ -173,7 +173,7 @@ TDNFPrepareAllPackages(
                dwError = TDNFIsFileOrSymlink(pszPkgName, &nIsFile);
                BAIL_ON_TDNF_ERROR(dwError);
 
-               if (nIsFile)
+               if (nIsFile && (fnmatch("*.rpm", pszPkgName, 0) == 0))
```

Comment:
should we print a warning or error if file is not rpm?

Reply:
I don't think so. If there is a file `glibc` and you type`tdnf install glibc`, tdnf will check but installing the package `glibc` (not the file) is almost certainly what you wanted.

Reply:
Understood. But in that case, `nIsFile` will be false right? So, you can move the fnmatch part inside the if and print error if not rpm.

Reply:
No, this is to cover the special case where `glibc`, in this example *is* a file, so `nIsFile` is true. And it's not an error - we just ignore the file and install `glibc` from the repositories.